packages/calcite-components/src/components/tile-select/tile-select.tsx

```diff
@@ -275,9 +275,11 @@ export class TileSelect implements InteractiveComponent, LoadableComponent {
   // --------------------------------------------------------------------------
 
   private renderInput(): void {
-    this.input = document.createElement(
-      this.type === "radio" ? "calcite-radio-button" : "calcite-checkbox",
-    );
+    this.input =
+      this.type === "radio"
+        ? /* we need to call createElement(x) separately to ensure supporting components are properly bundled */
+          document.createElement("calcite-radio-button")
```

Comment:
Looks good! 👍 

We might want to add this to a gotcha developer doc or add some kind of eslint for not having vars in createElement?

Reply:
Great idea. Will see if we can automate this. `import`s also share this issue.

Reply:
`import` is [already covered](https://github.com/Esri/calcite-design-system/blob/main/packages/calcite-components/.eslintrc.cjs#L77). Created https://github.com/Esri/calcite-design-system/issues/8651 for adding a custom rule to catch dynamic `createElement` tag usage.